tests/unit/modules/test_chroot.py

```diff
@@ -0,0 +1,281 @@
+            'files.rm_rf': MagicMock(),
+            'json.find_json': MagicMock(return_value={'return': {}})
+        }
+        salt_mock = {
+            'cmd.run': MagicMock(return_value=''),
+            'config.option': MagicMock(),
+            'cmd.run_chroot': MagicMock(return_value={
+                'retcode': 1,
+                'stderr': 'Error',
+            }),
+        }
+        with patch.dict(chroot.__utils__, utils_mock), \
+                patch.dict(chroot.__salt__, salt_mock):
+            self.assertEqual(chroot.call('/chroot', 'test.ping'), {
+                'result': False,
+                'comment': "Can't parse container command output"
+            })
+            utils_mock['thin.gen_thin'].assert_called_once()
+            salt_mock['config.option'].assert_called()
+            salt_mock['cmd.run'].assert_called_once()
+            salt_mock['cmd.run_chroot'].assert_called_with(
+                '/chroot',
+                ['python{}'.format(sys.version_info[0]), '/tmp01/salt-call',
+                 '--metadata', '--local',
+                 '--log-file', '/tmp01/log', '--cachedir', '/tmp01/cache',
+                 '--out', 'json', '-l', 'quiet', '--', 'test.ping'])
+            utils_mock['files.rm_rf'].assert_called_once()
+
+    @patch('salt.modules.chroot.exist')
+    @patch('tempfile.mkdtemp')
+    def test_call_success(self, mkdtemp, exist):
+        '''
+        Test execution of Salt functions in chroot.
+        '''
+        # Success test
+        exist.return_value = True
+        mkdtemp.return_value = '/chroot/tmp01'
+        utils_mock = {
+            'thin.gen_thin': MagicMock(return_value='/salt-thin.tgz'),
+            'files.rm_rf': MagicMock(),
+            'json.find_json': MagicMock(return_value={'return': 'result'})
+        }
+        salt_mock = {
+            'cmd.run': MagicMock(return_value=''),
+            'config.option': MagicMock(),
+            'cmd.run_chroot': MagicMock(return_value={
+                'retcode': 0,
+                'stdout': '',
+            }),
+        }
+        with patch.dict(chroot.__utils__, utils_mock), \
+                patch.dict(chroot.__salt__, salt_mock):
+            self.assertEqual(chroot.call('/chroot', 'test.ping'), 'result')
+            utils_mock['thin.gen_thin'].assert_called_once()
+            salt_mock['config.option'].assert_called()
+            salt_mock['cmd.run'].assert_called_once()
+            salt_mock['cmd.run_chroot'].assert_called_with(
+                '/chroot',
+                ['python{}'.format(sys.version_info[0]), '/tmp01/salt-call',
+                 '--metadata', '--local',
+                 '--log-file', '/tmp01/log', '--cachedir', '/tmp01/cache',
+                 '--out', 'json', '-l', 'quiet', '--', 'test.ping'])
+            utils_mock['files.rm_rf'].assert_called_once()
+
+    @patch('salt.modules.chroot.exist')
+    @patch('tempfile.mkdtemp')
+    def test_call_success_parameters(self, mkdtemp, exist):
```

Comment:
This test is failing on some OSes:
```Traceback (most recent call last):
  File "/tmp/kitchen/testing/.nox/runtests-parametrized-3-coverage-true-crypto-none-transport-zeromq/lib/python3.5/site-packages/mock/mock.py", line 1330, in patched
    return func(*args, **keywargs)
  File "/tmp/kitchen/testing/tests/unit/modules/test_chroot.py", line 236, in test_call_success_parameters
    '--', 'ssh.set_auth_key', 'user=user', 'key=key'])
  File "/tmp/kitchen/testing/.nox/runtests-parametrized-3-coverage-true-crypto-none-transport-zeromq/lib/python3.5/site-packages/mock/mock.py", line 944, in assert_called_with
    six.raise_from(AssertionError(_error_message(cause)), cause)
  File "<string>", line 3, in raise_from
AssertionError: expected call not found.
Expected: mock('/chroot', ['python3', '/tmp01/salt-call', '--metadata', '--local', '--log-file', '/tmp01/log', '--cachedir', '/tmp01/cache', '--out', 'json', '-l', 'quiet', '--', 'ssh.set_auth_key', 'user=user', 'key=key'])
Actual: mock('/chroot', ['python3', '/tmp01/salt-call', '--metadata', '--local', '--log-file', '/tmp01/log', '--cachedir', '/tmp01/cache', '--out', 'json', '-l', 'quiet', '--', 'ssh.set_auth_key', 'key=key', 'user=user'])
```

Reply:
It looks like `key=key` and `user=user` changed places on the actual return

Reply:
Ah this is tricky, as the ordering of kwargs makes the ordering unpredictable. This tests comes from https://github.com/saltstack/salt/pull/54959. I will figure a fix there and rebase this commit on top of the other one.

Thanks a lot!

Reply:
Done